qcodes/instrument_drivers/Keysight/keysightb1500/KeysightB1500_base.py

```diff
@@ -43,6 +47,66 @@ def __init__(self, name, address, **kwargs):
             Set the function to OFF in cases that the measurement speed is 
             more important than the measurement accuracy. This roughly halves
             the integration time."""))
+
+        self.add_parameter('use_nplc_for_high_speed_adc',
+                           set_cmd=self._set_nplc_for_high_speed_adc,
+                           get_cmd=self._get_nplc_for_high_speed_adc,
+                           vals=vals.Numbers(0, 1023),
+                           doctsring=textwrap.dedent("""
+            Set the high-speed ADC to NPLC mode, with optionally defining 
+            number of averaging samples via argument `n`.
+
+            Args:
+                n: Value that defines the number of averaging samples given by
+                    the following formula:
+
+                    ``Number of averaging samples = n / 128``.
+
+                    n=1 to 100. Default setting is 1 (if `None` is passed).
+
+                    The Keysight B1500 gets 128 samples in a power line cycle,
+                    repeats this for the times you specify, and performs
+                    averaging to get the measurement data. (For more info see
+                    Table 4-21.).  Note that the integration time will not be
+                    updated if a non-integer value is written to the B1500.
+                    """))
+
+        self.add_parameter('use_nplc_for_high_resolution_adc',
+                           set_cmd=self._set_nplc_for_high_resolution_adc,
+                           get_cmd=self._get_nplc_for_high_resolution_adc,
+                           docstring=textwrap.dedent("""
+            Set the high-resolution ADC to NPLC mode, with optionally defining
+            the number of PLCs per sample via argument `n`.
+
+            Args:
+                n: Value that defines the integration time given by the
+                    following formula:
+    
+                    ``Integration time = n / power line frequency``.
+    
+                    n=1 to 100. Default setting is 1 (if `None` is passed).
+                    (For more info see Table 4-21.).  Note that the integration
+                    time will not be updated if a non-integer value is written
+                    to the B1500."""))
+
+        self.add_parameter('use_manual_mode_for_high_speed_adc',
+                           set_cmd=self._set_manual_mode_for_high_speed_adc,
+                           get_cmd=self._get_manual_mode_for_high_speed_adc,
```

Comment:
Missing validator

Reply:
This validator is tough to set (mode =3 in the above picture). It depends on lot of conditions so I left it as is.

Reply:
Can we not set it to be a float between 2e-6 and 20e-3 ? That will not capture the resolution but is better than nothing

Reply:
I did not read it properly. Thanks :)